src/bindings/python/flux/job/watcher.py

```diff
@@ -0,0 +1,574 @@
+        self.labelio = labelio
+        self.exitcode = 0
+        self.show_progress = progress
+        self.progress = JobProgressBar(flux_handle, starttime=self.t0, jps=jps)
+
+        self._states = Counter()
+
+        if jobs:
+            self.add_jobs(*jobs)
+
+    @staticmethod
+    def _status_to_exitcode(status):
+        """Calculate exitcode from job status"""
+        if os.WIFEXITED(status):
+            status = os.WEXITSTATUS(status)
+        elif os.WIFSIGNALED(status):
+            status = 127 + os.WTERMSIG(status)
+        return status
+
+    def start(self):
+        """
+        Start JobWatcher progress bar if configured
+        """
+        if self.show_progress:
+            self.progress.start()
+        return self
+
+    def stop(self):
+        """
+        Stop JobWatcher progress bar if configured
+        """
+        if self.show_progress:
+            self.progress.stop()
+        return self
+
+    def add_jobs(self, *jobs, stdout=None, stderr=None, wait="clean"):
+        """
+        Begin monitoring the progress of a set of jobs
+
+        Args:
+            *jobs (JobID): one or more jobs to monitor
+            stdout (TextIOWrapper): destination for stdout for this job
+                or jobs. If None, use default for this JobWatcher instance.
+            stderr: (TextIOWrapper): destination for stderr for this job
+                or jobs (if None, then set to same location as ``stdout``)
+            wait: event at which to stop watching the job (default=clean)
+        """
+
+        self.progress.add_jobs(*jobs)
+
+        if stdout is None:
+            stdout = self.stdout
+        if stderr is None:
+            stderr = self.stderr
+
+        for job in jobs:
+            if not self.t0 or job.t_submit < self.t0:
+                self.t0 = job.t_submit
+
+            job_status = self.JobWatchStatus(job, stdout, stderr, wait=wait)
+            flux.job.event_watch_async(self.flux_handle, job.id).then(
+                self._event_watch_cb, job_status
+            )
+            if not job_status.active:
+                self._progress_update(job_status)
+        return self
+
+    def add_jobid(self, jobid, stdout=None, stderr=None, wait="clean"):
+        """
+        Begin monitoring the progress of a job by jobid
+
+        Args:
+            jobid (JobID): one or more jobs to monitor
+            stdout (TextIOWrapper): destination for stdout for this job
+                or jobs. If None, use default for this JobWatcher instance.
+            stderr: (TextIOWrapper): destination for stderr for this job
+                or jobs (if None, then set to same location as ``stdout``)
+            wait: event at which to stop watching the job (default=clean)
+        """
+        job = flux.job.JobInfo(
+            {
+                "id": jobid,
+                "state": flux.constants.FLUX_JOB_STATE_SCHED,
+                "t_submit": time.time(),
+            }
+        )
```

Comment:
i think to a casual observer this looks sort of odd, to be hardcoding this stuff.  I guess what's going on is gotta initialize to something, then the event watcher will take care of the rest.  Perhaps a small comment would suffice.